src/gridcoin/scraper/scraper.cpp

```diff
@@ -2721,6 +2725,18 @@ bool ProcessProjectRacFileByCPID(const std::string& project, const fs::path& fil
                 }
             }
 
+            // We need to accumulate total credit across ALL project cpids, regardless of their beacon status, to get
+            // an "all cpid" total credit sum to be used for automatic greylisting purposes.
+            std::string s_cpid_total_credit = ExtractXML(data, "<total_credit>", "</total_credit>");
+            double cpid_total_credit = 0;
+
+            if (!ParseDouble(s_cpid_total_credit, &cpid_total_credit)) {
+                _log(logattribute::ERR, __func__, "Bad cpid total credit in user stats file data.");
+                continue;
+            }
+
+            all_cpid_total_credit += cpid_total_credit;
```

Comment:
(re: magnitude calculation) Are non-this PR credit parsing using doubles as well?

Reply:
Everything in the scraper uses doubles, and always has. The double to int64 occurs downstream of the scraper code. I left it that way to be consistent.